docs/libraries/azure-resource-manager/rules/no-record.md

```diff
@@ -0,0 +1,67 @@
+---
+title: no-record
+---
+
+```text title=- Full name-
+@azure-tools/typespec-azure-resource-manager/no-record
+```
+
+ARM requires Resource provider teams to define types explicitly. This is to ensure good customer experience in terms of the discoverability of concrete type definitions.
+
+#### ❌ Incorrect
+
+```tsp
+model Address extends Record<string> {}
```

Comment:
Nit, would put the `is Record<T>` example first as this is recommended over `extends`